test/libyul/yulOptimizerTests/unusedAssignEliminator/for_deep_noremove.yul

```diff
@@ -46,10 +46,7 @@
 //                         for { } 1 { }
 //                         {
 //                             for { } 1 { a := 10 }
-//                             {
-//                                 b := 12
-//                                 b := 11
-//                             }
+//                             { b := 11 }
```

Comment:
Well, now this test case isn't really testing what its name claims to test anymore.

Reply:
Actually also not quite clear on why this changes, so would be good to clarify this. Apart from that being a bit suspicious, the PR generally looks correct to me.

Reply:
It does: You can remove the `b := 11` if you do the full analysis. The thing that changed is that it does not blindly mark everything as used, but please double-check!

Reply:
Right, true - for the record: if this had fewer loop nestings, even the ``b := 11`` could be removed, since it's not ever used later on. But the loop body can still be analyzed in isolation. Still need to double-check this in the code, but the test *does* make sense, so resolving this.